### PR TITLE
Add initial version of runtimetest command:

### PR DESCRIPTION
**Pull request by zenlinTechnofreak**
Add initial version of runtimetest command:
add unit and config for managing cases;
replace runtime_test.sh to golang realization;

Signed-off-by: zen Lin(Lin Zhinan) linzhinan@huawei.com

### CODE REVIEW

**Comment by mrunalp** [comment]
Needs rebase. (On my review list).

**Comment by zenlinTechnofreak** [comment]
@mrunalp I have reabsed it, and this is a initial version of runtimetest sub command, let us talk about 'add more test cases and support other runtimes if they have wrappers to CLI similar to runc.
' in the issuse Initial code of runtimetest command #10 for more progress.

**Comment by zenlinTechnofreak** [comment]
@mrunalp  Any review comments plz let me know, thx.

**Comment by zenlinTechnofreak** [comment]
@liangchenye I have already reabased the PR.

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	if err != nil {
+		logrus.Fatalf("open file %v error %v", configPath, err)
+	}
+	defer f.Close()
+
+	rd := bufio.NewReader(f)
+	count := 0
+
+	for {
+
```
Comment (mrunalp) [inline config/config.go]: unnecessary empty line here.

File: config/config.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+func init() {
+	f, err := os.Open(configPath)
+	if err != nil {
+		logrus.Fatalf("open file %v error %v", configPath, err)
+	}
+	defer f.Close()
+
+	rd := bufio.NewReader(f)
+	count := 0
+
```
Comment (mrunalp) [inline config/config.go]: could club the initialization and for loop together without an empty line

File: config/config.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		for i, arg := range splitArgs(caseArg) {
+			BundleMap[caseName+strconv.FormatInt(int64(i), 10)] = arg
+			count = count + 1
+		}
+	}
+	configLen = count
+}
+
+func splitArgs(args string) []string {
+
```
Comment (mrunalp) [inline config/config.go]: empty line not required here

File: runtimetest.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/codegangsta/cli"
+	"github.com/opencontainers/ocitools/units"
+)
+
+const bundleCacheDir = "./bundles"
+
+var runtimetestFlags = []cli.Flag{
+	cli.StringFlag{Name: "runtime, r", Usage: "runtime to be tested"},
+	cli.StringFlag{Name: "output, o", Usage: "output format, \n" +
+		"-o=all: ouput sucessful details and statics, -o=err-only: ouput failure details and statics"},
```
Comment (mrunalp) [inline runtimetest.go]: s/statics/statistics?

File: runtimetest.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	cli.BoolFlag{Name: "debug, d", Usage: "switch of debug mode, defaults to false, with '--debug' to enable debug mode"},
+}
+
+var runtimeTestCommand = cli.Command{
+	Name:  "runtimetest",
+	Usage: "test if a runtime is comlpliant to oci specs",
+	Flags: runtimetestFlags,
+	Action: func(context *cli.Context) {
+
```
Comment (mrunalp) [inline runtimetest.go]: empty line not required.

File: runtimetest.go [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		if os.Geteuid() != 0 {
+			logrus.Fatalln("runtimetest should be run as root")
+		}
+		var runtime string
+		if runtime = context.String("runtime"); runtime != "runc" {
+			logrus.Fatalf("runtimetest have not support %v\n", runtime)
```
Comment (mrunalp) [inline runtimetest.go]: s/have not/does not/

**Comment by zenlinTechnofreak** [comment]
@mrunalp  Thanks very much, I have reviewed it myself and also fixed the format problems you pointed out.

**Comment by liangchenye** [comment]
Looks good, I have three suggestions:
1.  less new directory (config and units)
  maybe  merge ocitools/config/config.go and ocitools/units/units.go into ocitools/runtimetest.go 
2. fix typos
  although my English grammar is pool, I try to comment on new commits.
3. less commits
  merge some commits together?

**Comment by wking** [comment]
Close now that #336 has landed?

**Comment by Mashimiao** [comment]
as #336  landed, close this

**Pull request closed without merging**